### PR TITLE
Add `From<MatZ> for MatPolyOverZ`

### PR DESCRIPTION
**Pull request by MoogSven**
Note : Requires rebase on dev once #318 is merged.

**Description**

This PR implements...
- [x] Add `From<MatZ> for MatPolyOverZ`

**Testing**

- [x] I added basic working examples (possibly in doc-comment)
- [x] I added tests for large (pointer representation) values
- [x] I triggered all possible errors in my test in every possible way
- [x] I included tests for all reasonable edge cases

**Checklist:**

- [x] I have performed a self-review of my own code
  - [x] The code provides good readability and maintainability s.t. it fulfills best practices like talking code, modularity, ...
    - [x] The chosen implementation is not more complex than it has to be
  - [x] My code should work as intended and no side effects occur (e.g. memory leaks)
  - [x] The doc comments fit our style guide
  - [x] I have credited related sources if needed

### CODE REVIEW

File: src/integer/mat_poly_over_z/from.rs [hunk 1]
```diff
@@ -194,3 +230,41 @@ mod test_from_str {
...
+    fn small() {
+        let matz_str = "[[1,2,3],[4,5,6]]";
+        let matz = MatZ::from_str(matz_str).unwrap();
+
+        let mat_poly = MatPolyOverZ::from(&matz);
+
+        let poly_mat_cmp_str = "[[1  1, 1  2, 1  3],[1  4, 1  5, 1  6]]";
+        let mat_poly_cmp = MatPolyOverZ::from_str(poly_mat_cmp_str).unwrap();
+        assert_eq!(mat_poly, mat_poly_cmp);
+    }
```
Comment (Marcel583) [inline src/integer/mat_poly_over_z/from.rs:247]: Add comments for test cases

**Review by Marcel583 — APPROVED** [review]

File: src/integer/mat_poly_over_z/from.rs (hunk 1 above)
Comment (MoogSven) [reply src/integer/mat_poly_over_z/from.rs]: Added

File: src/integer/mat_poly_over_z/from.rs [hunk 2]
```diff
@@ -91,6 +91,42 @@ impl FromStr for MatPolyOverZ {
...
+    /// ```
+    /// use qfall_math::integer::{MatZ, MatPolyOverZ};
+    ///
+    /// let mat_z = MatZ::identity(10,10);
+    /// let mat_poly = MatPolyOverZ::from(&mat_z);
+    /// ```
+    fn from(constants: &MatZ) -> Self {
+        let num_rows = constants.get_num_rows();
+        let num_columns = constants.get_num_columns();
+        let mut out = MatPolyOverZ::new(num_rows, constants.get_num_columns());
```
[the remark this answers is not in the document]
Comment (MoogSven) [reply src/integer/mat_poly_over_z/from.rs]: Added

File: src/integer/mat_poly_over_z/from.rs [hunk 3]
```diff
@@ -91,6 +91,42 @@ impl FromStr for MatPolyOverZ {
...
+    /// set to the value in the constants matrix.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::{MatZ, MatPolyOverZ};
+    ///
+    /// let mat_z = MatZ::identity(10,10);
+    /// let mat_poly = MatPolyOverZ::from(&mat_z);
+    /// ```
+    fn from(constants: &MatZ) -> Self {
```
[the remark this answers is not in the document]
Comment (MoogSven) [reply src/integer/mat_poly_over_z/from.rs]: Changed to "matrix"

**Pull request merged**